sled-agent/src/bootstrap/server.rs

```diff
@@ -63,6 +66,23 @@ impl Server {
             debug!(log, "registered DTrace probes");
         }
 
+        // Turn on the maghemite routing service.
+        info!(log, "Starting mg-ddm service");
+        let link = sled_config
+            .get_link()
+            .map_err(|err| format!("Failed to find physical link: {err}"))?;
+
+        // TODO-correctness (a) Is "linklocal" always right? (b) Do we need
```

Comment:
Sorry, I should've been more specific. IIUC the _name_ `"linklocal"` is just convention and not special, but maybe that's good enough? E.g., when I create fresh VMs, I don't actually have a `$PHYS/linklocal` address object without creating one manually.

The breakage is that starting the service fails if the address object we specify here doesn't exist. I think that failure is correct (since we won't be able to route off node without mg-ddm), but I don't want it to fail based on a bad assumption about this name.

Reply:
We definitely have to have a link-local address, yes. I don't know who or when should create that - should bootstrap-agent do that here if doesn't exist?

Reply:
I'm not sure as to who creates the address, but I believe the intention is for that to be over the loopback interface `lo0`. (See [this](https://rfd.shared.oxide.computer/rfd/0259#_bootstrap_address_assignment) for some notes.)

There are multiple Chelsio physical links on the sled, both of which provide a potential path to talk to the bootstrap server. Maghemite is responsible for advertising the prefix over both Chelio links, and updating the OS's routing tables accordingly. Although @rcgoodfellow can talk in much more depth about it, my understanding is that traffic will be dynamically routed through one or the other Chelsio depending on the expected delay sending a packet down each of them. But it expects to be able to reach the bootstrap address through either link.

Reply:
Maghemite needs link-local/addrconf addresses to be set up for both sled physical interfaces it will route over. So if our interfaces on the sled are

- cxgbe0
- cxgbe1

Then the addrconf addresses 

- cxgbe0/linklocal
- cxgbe1/linklocal

should be set up (not advocating specifically for "linklocal" there, but that seems to be the trend so rolling with it, I don't have a good name to suggest) early in the sled initialization process.

I don't have an opinion on who sets those up, only that they are set up before Maghemite starts so the address object names (i.e. "cxgbe0/linklocal") can be [provided to Maghemite on startup](https://github.com/oxidecomputer/testbed/blob/main/halfstack-2x2/cargo-bay/03-init-router.sh#L6). If the addrconf addresses are set up as temporary, this ordering of addresses before Maghemite will need to be ensured on each boot.

Reply:
Poking around, we don't need to extend `AddressRequest` at all; we already have a function that will ensure an `AddrObject` exists with a link-local address. Added a call to it in 6309c7bb291ef019e3eda31e2ff644599b5793ef. I believe this will fail if the interface already has an `addrconf` address of a different name, but hopefully that's okay? If something else is creating addrconf addresses before sled-agent, we should sync up on the name.